### PR TITLE
fix match step range

### PR DESCRIPTION
**Pull request by nevermore3**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
close https://github.com/vesoft-inc/nebula-ent/issues/2045
#### Description:
match (v:player)-[e:like*2]->(b) where id(v) == 'Tony Parker' AND (b)-[*1..2]->(v) return distinct e

before

```
|  8 | Traverse       | 7            |                | outputVar: {                                        |
|    |                |              |                |   "colNames": [                                     |
|    |                |              |                |     "b",                                            |
|    |                |              |                |     "__VAR_1"                                       |
|    |                |              |                |   ],                                                |
|    |                |              |                |   "type": "DATASET",                                |
|    |                |              |                |   "name": "__Traverse_8"                            |
|    |                |              |                | }                                                   |
|    |                |              |                | inputVar: __Argument_7                              |
|    |                |              |                | space: 6                                            |
|    |                |              |                | dedup: true                                         |
|    |                |              |                | limit: -1                                           |
|    |                |              |                | filter:                                             |
|    |                |              |                | orderBy: []                                         |
|    |                |              |                | src: id($-.b)                                       |
|    |                |              |                | edgeTypes: []                                       |
|    |                |              |                | edgeDirection: OUT_EDGE                             |
|    |                |              |                | vertexProps:                                        |
|    |                |              |                | edgeProps: [                                        |
|    |                |              |                |   {                                                 |
|    |                |              |                |     "props": [                                      |
|    |                |              |                |       "_dst",                                       |
|    |                |              |                |       "_type",                                      |
|    |                |              |                |       "_rank"                                       |
|    |                |              |                |     ],                                              |
|    |                |              |                |     "type": 12                                      |
|    |                |              |                |   },                                                |
|    |                |              |                |   {                                                 |
|    |                |              |                |     "type": 10,                                     |
|    |                |              |                |     "props": [                                      |
|    |                |              |                |       "_dst",                                       |
|    |                |              |                |       "_type",                                      |
|    |                |              |                |       "_rank"                                       |
|    |                |              |                |     ]                                               |
|    |                |              |                |   },                                                |
|    |                |              |                |   {                                                 |
|    |                |              |                |     "type": 11,                                     |
|    |                |              |                |     "props": [                                      |
|    |                |              |                |       "_dst",                                       |
|    |                |              |                |       "_type",                                      |
|    |                |              |                |       "_rank"                                       |
|    |                |              |                |     ]                                               |
|    |                |              |                |   }                                                 |
|    |                |              |                | ]                                                   |
|    |                |              |                | statProps:                                          |
|    |                |              |                | exprs:                                              |
|    |                |              |                | random: false                                       |
|    |                |              |                | steps: 0..140233306976544                           |
|    |                |              |                | vertex filter:                                      |
|    |                |              |                | edge filter:                                        |
|    |                |              |                | if_track_previous_path: false                       |
|    |                |              |                | first step filter:                                  |
```
After

```
|  8 | Traverse       | 7            |                | outputVar: {                                        |
|    |                |              |                |   "colNames": [                                     |
|    |                |              |                |     "b",                                            |
|    |                |              |                |     "__VAR_1"                                       |
|    |                |              |                |   ],                                                |
|    |                |              |                |   "type": "DATASET",                                |
|    |                |              |                |   "name": "__Traverse_8"                            |
|    |                |              |                | }                                                   |
|    |                |              |                | inputVar: __Argument_7                              |
|    |                |              |                | space: 11                                           |
|    |                |              |                | dedup: true                                         |
|    |                |              |                | limit: -1                                           |
|    |                |              |                | filter:                                             |
|    |                |              |                | orderBy: []                                         |
|    |                |              |                | src: id($-.b)                                       |
|    |                |              |                | edgeTypes: []                                       |
|    |                |              |                | edgeDirection: OUT_EDGE                             |
|    |                |              |                | vertexProps:                                        |
|    |                |              |                | edgeProps: [                                        |
|    |                |              |                |   {                                                 |
|    |                |              |                |     "type": 17,                                     |
|    |                |              |                |     "props": [                                      |
|    |                |              |                |       "_dst",                                       |
|    |                |              |                |       "_type",                                      |
|    |                |              |                |       "_rank"                                       |
|    |                |              |                |     ]                                               |
|    |                |              |                |   },                                                |
|    |                |              |                |   {                                                 |
|    |                |              |                |     "props": [                                      |
|    |                |              |                |       "_dst",                                       |
|    |                |              |                |       "_type",                                      |
|    |                |              |                |       "_rank"                                       |
|    |                |              |                |     ],                                              |
|    |                |              |                |     "type": 15                                      |
|    |                |              |                |   },                                                |
|    |                |              |                |   {                                                 |
|    |                |              |                |     "props": [                                      |
|    |                |              |                |       "_dst",                                       |
|    |                |              |                |       "_type",                                      |
|    |                |              |                |       "_rank"                                       |
|    |                |              |                |     ],                                              |
|    |                |              |                |     "type": 16                                      |
|    |                |              |                |   }                                                 |
|    |                |              |                | ]                                                   |
|    |                |              |                | statProps:                                          |
|    |                |              |                | exprs:                                              |
|    |                |              |                | random: false                                       |
|    |                |              |                | steps: 1..2                                         |
|    |                |              |                | vertex filter:                                      |
|    |                |              |                | edge filter:                                        |
|    |                |              |                | if_track_previous_path: false                       |
|    |                |              |                | first step filter:                                  |
|    |                |              |                | tag filter:                                         |
```
The steps in traverse used to be in the pointer format, and it was shallow copied when copying, causing the content pointed to by the steps to be released early

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/graph/context/ast/CypherAstContext.h [hunk 1]
```diff
@@ -46,7 +46,7 @@ struct NodeInfo {
 
 struct EdgeInfo {
   bool anonymous{false};
-  MatchStepRange* range{nullptr};
```
Comment (Shylock-Hg) [inline src/graph/context/ast/CypherAstContext.h:49]: As I known, this AST is released when query executed.

Comment (nevermore3) [reply src/graph/context/ast/CypherAstContext.h:49]: Isn't it in the memory pool?

Comment (Shylock-Hg) [reply src/graph/context/ast/CypherAstContext.h:49]: All AST will hold by `QueryContext`.

Comment (nevermore3) [reply src/graph/context/ast/CypherAstContext.h:49]: AST information will be transferred to Sentence in validate phase.  the sentence is saved in the querycontext during the execution phase,. The current problem is the wild pointer caused by the release of the temporary variable

**Review by yixinglu — DISMISSED** [review]

File: src/graph/context/ast/CypherAstContext.h (hunk 1 above)
Comment (Shylock-Hg) [reply src/graph/context/ast/CypherAstContext.h:49]: Better change https://github.com/vesoft-inc/nebula/blob/fdca644c7be90719cddfb10cb0735f0b7533136c/src/graph/validator/MatchValidator.cpp#L1180 to `std::vector<MatchPath*>`, we assume lifetime of AST is available before complete query execution.

**Review by Shylock-Hg — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**